app/src/main/java/com/example/sharewhatyoucanproject/donor/DonorActivity.kt

```diff
@@ -0,0 +1,155 @@
+                    finish()
+                } else {
+                    circularProgressIndicator.visibility = View.GONE
+                    applicationContext.showToast(it)
+                }
+            }
+        }
+
+        binding.foodimg.setOnClickListener {
+            val intent = Intent()
+            intent.type = "image/*"
+            intent.action = Intent.ACTION_GET_CONTENT
+            startActivityForResult(Intent.createChooser(intent, "Select Picture"), 0)
+        }
+
+        arrayAdapter = ArrayAdapter<String>(
+            applicationContext,
+            android.R.layout.simple_spinner_dropdown_item,
+            foodTypes,
+        )
+        binding.spinner.adapter = arrayAdapter
+        binding.spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View,
+                position: Int,
+                id: Long,
+            ) {
+                foodTypes[position].also {
+                    donorViewModel.foodType = it
+                    applicationContext.showToast(it)
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+
+        binding.uploadbtn.setOnClickListener {
+            donorViewModel.title = binding.titleet.text.toString()
+            donorViewModel.description = binding.descet.text.toString()
+            donorViewModel.hourSet = binding.hourset.text.toString()
+
+            if (!donorViewModel.isDataCompleted()) {
+                applicationContext.showToast("Please Fill All Fields")
+                return@setOnClickListener
+            }
+
+            if (donorViewModel.isImageNotSelected()) {
+                applicationContext.showToast("Choose a image")
+                return@setOnClickListener
+            }
+
+            if (!hasPermissions(applicationContext, *locationPermissions)) {
+                applicationContext.showToast("Please Allow Location To Post")
+                ActivityCompat.requestPermissions(
+                    this@DonorActivity,
+                    locationPermissions,
+                    PERMISSION_ALL,
+                )
+                return@setOnClickListener
+            }
+
+            if (!isGPSEnabled(applicationContext)) {
+                applicationContext.showToast("Open Your GPS")
+                return@setOnClickListener
+            }
+
+            circularProgressIndicator.visibility = View.VISIBLE
+            if (donorViewModel.isFoodNotValid()) {
+                circularProgressIndicator.visibility = View.GONE
+                applicationContext.showToast("You Cannot Upload Food")
+                return@setOnClickListener
+            }
+
+            donorViewModel.getCurrentLocation()
+        }
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
`onActivityResult` is deprecated. Here are the docs for the new pattern: https://developer.android.com/training/basics/intents/result. Let me know if you need help with this, it's pretty different from the old pattern.